app.js

```diff
@@ -287,12 +317,26 @@ router.post('/account/sendRecoveryMessage', async ctx => {
     }
 
     const account = await models.Account.findOne({ where: { accountId } });
-    const [recoveryMethod] = await account.getRecoveryMethods({ where: {
-        kind: method.kind,
-        detail: method.detail
-    }});
 
-    await recoveryMethod.update({ publicKey, securityCode: null });
+    if (isNew) {
+        // remove all recovery methods
+        const allRecoveryMethods = await account.getRecoveryMethods();
+        for (const rm of allRecoveryMethods) {
+            await rm.destroy();
+        }
+        // set up the official recovery method for this account
```

Comment:
what do you mean by "official" method?

Reply:
using isNew flag in body, we will clear all recovery methods and set up this one. Changed "official" to "first".